commit d7cb764bc10756d1535d5d8d9b486dec13dd4ca9
Author: Michael Still <mikal@stillhq.com>
Date:   Thu Jul 3 13:52:30 2025 +1000

    Add syslog to fluentd.

diff --git a/docker/fluentd/Dockerfile.j2 b/docker/fluentd/Dockerfile.j2
index 899b4b6..5cf14a9 100644
--- a/docker/fluentd/Dockerfile.j2
+++ b/docker/fluentd/Dockerfile.j2
@@ -48,6 +48,7 @@ COPY extend_start.sh /usr/local/bin/kolla_extend_start
     'fluent-plugin-grok-parser',
     'fluent-plugin-prometheus',
     'fluent-plugin-rewrite-tag-filter',
+    'fluent-plugin-remote-syslog',
 ] %}
 
 {{ macros.install_fluent_plugins(fluentd_plugins | customizable("plugins")) }}
